### PR TITLE
Add command for exporting all available channels in a guild for client

### PR DESCRIPTION
**Pull request by TVdata**
New command "exportguild" that is used the same way the "export" command is but exports all available channels within a guild. Id is given using "-g --guild". This is probably pretty crude and should be improved, but this is the solution I'm currently using.

### CODE REVIEW

File: DiscordChatExporter.Cli/DiscordChatExporter.Cli.csproj [hunk 1]
```diff
@@ -7,6 +7,7 @@
         <Company>Tyrrrz</Company>
         <Copyright>Copyright (c) 2017-2018 Alexey Golub</Copyright>
         <ApplicationIcon>..\favicon.ico</ApplicationIcon>
+        <NoWin32Manifest>true</NoWin32Manifest>
```
Comment (Tyrrrz) [inline DiscordChatExporter.Cli/DiscordChatExporter.Cli.csproj]: What's the purpose of this?

File: DiscordChatExporter.Cli/Verbs/ExportGuildVerb.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            if (Options.DateFormat.IsNotBlank())
+                settingsService.DateFormat = Options.DateFormat;
+            if (Options.MessageGroupLimit > 0)
+                settingsService.MessageGroupLimit = Options.MessageGroupLimit;
+
+            // Get channels
+            var channels = await dataService.GetGuildChannelsAsync(Options.GetToken(), Options.GuildId);
+
+            // Loop through channels
+            foreach (var channel in channels.Where(c => c.Type.IsEither(ChannelType.GuildTextChat))
```
Comment (Tyrrrz) [inline DiscordChatExporter.Cli/Verbs/ExportGuildVerb.cs]: no reason to use `.IsEither` when comparing to only one instance

File: DiscordChatExporter.Cli/Verbs/ExportGuildVerb.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                .OrderBy(c => c.Name))
```
Comment (Tyrrrz) [inline DiscordChatExporter.Cli/Verbs/ExportGuildVerb.cs]: is it important to order here?

File: DiscordChatExporter.Core/Services/DataService.cs [hunk 4]
```diff
@@ -63,6 +63,38 @@ public partial class DataService : IDisposable
             });
         }
 
+        public async Task<bool> GetAccessAsync(AuthToken token, string resource, string endpoint)
+        {
+            // Create request policy
```
Comment (Tyrrrz) [inline DiscordChatExporter.Core/Services/DataService.cs]: I would rewrite this to just try `GetApiResponseAsync` and catch `HttpErrorStatusCodeException` with status code 403.
Also, rename the method to `CheckAccessAsync` maybe.

File: Readme.md [hunk 5]
```diff
@@ -1,29 +1,6 @@
...
-- Supports both user tokens and bot tokens
-- Allows retrieving messages in specified date range
-- Multiple export formats: HTML (dark/light), TXT and CSV
-- Renders all message features including: markdown, attachments, embeds, emojis, mentions, etc
-
-## Screenshots
-
-![](http://www.tyrrrz.me/Projects/DiscordChatExporter/Images/1.png)
-![](http://www.tyrrrz.me/Projects/DiscordChatExporter/Images/4.png)
+Fork of Tyrrz DiscordChatExporter. New "exportguild" command in CLI.
```
Comment (Tyrrrz) [inline Readme.md]: If this is to be merged, please don't delete the readme :D

**Comment by Tyrrrz** [comment]
Cool! Added some review comments

File: DiscordChatExporter.Core/Services/DataService.cs (hunk 4 above)
Comment (Tyrrrz) [reply DiscordChatExporter.Core/Services/DataService.cs]: Or just not use it and have a try/catch in the command `ExecuteAsync` which swallows 403 exceptions.

File: DiscordChatExporter.Cli/DiscordChatExporter.Cli.csproj (hunk 1 above)
Comment (TVdata) [reply DiscordChatExporter.Cli/DiscordChatExporter.Cli.csproj]: Automated change by visual studio, I'll remove it.

**Comment by TVdata** [comment]
Done  @Tyrrrz

**Comment by Tyrrrz** [comment]
Appears that the build was broken:
`DiscordChatExporter.Gui -> C:\projects\discordchatexporter\DiscordChatExporter.Gui\bin\Release\DiscordChatExporter.exe
Verbs\ExportGuildVerb.cs(38,56): error CS1061: 'DataService' does not contain a definition for 'GetApiResponseAsync' and no accessible extension method 'GetApiResponseAsync' accepting a first argument of type 'DataService' could be found (are you missing a using directive or an assembly reference?) `

**Pull request merged**